executor/aggfuncs/func_count_distinct.go

```diff
@@ -254,6 +255,7 @@ func (e *countOriginalWithDistinct4String) UpdatePartialResult(sctx sessionctx.C
 		if p.valSet.Exist(input) {
 			continue
 		}
+		input = stringutil.Copy(input)
```

Comment:
how about modifying the `stringmap`?

Reply:
```sh
[jianzhang.zj:~/Code/tidb] git:(master ✔)
➜ grepgo "set.StringSet" | grep -v "test" | grep -v "(" | grep -v "{" | grep "set.StringSet"
./planner/core/planbuilder.go:354:      underlyingViewNames set.StringSet
./planner/core/memtable_predicate_extractor.go:183:     result set.StringSet,
./planner/core/memtable_predicate_extractor.go:472:     excludeCols set.StringSet,
./planner/core/memtable_predicate_extractor.go:482:             var values set.StringSet
./planner/core/memtable_predicate_extractor.go:513:     NodeTypes set.StringSet
./planner/core/memtable_predicate_extractor.go:519:     Instances set.StringSet
./planner/core/memtable_predicate_extractor.go:566:     NodeTypes set.StringSet
./planner/core/memtable_predicate_extractor.go:572:     Instances set.StringSet
./planner/core/memtable_predicate_extractor.go:585:     LogLevels set.StringSet
./planner/core/memtable_predicate_extractor.go:668:     LabelConditions map[string]set.StringSet
./planner/core/memtable_predicate_extractor.go:775:     MetricsNames set.StringSet
./planner/core/memtable_predicate_extractor.go:805:     Rules set.StringSet
./planner/core/memtable_predicate_extractor.go:808:     Items set.StringSet
./planner/core/memtable_predicate_extractor.go:838:     Rules       set.StringSet
./planner/core/memtable_predicate_extractor.go:839:     MetricNames set.StringSet
./planner/core/memtable_predicate_extractor.go:872:     Types       set.StringSet
./planner/core/util.go:242:// extractStringFromStringSet helps extract string info from set.StringSet
./executor/aggregate.go:86:     groupSet            set.StringSet
./executor/aggregate.go:148:    groupSet         set.StringSet
./executor/aggfuncs/func_count_distinct.go:122: valSet set.StringSet
./executor/aggfuncs/func_count_distinct.go:218: valSet set.StringSet
./executor/aggfuncs/func_count_distinct.go:268: valSet set.StringSet
./executor/aggfuncs/func_group_concat.go:155:   valSet            set.StringSet
./executor/aggfuncs/func_group_concat.go:427:   valSet            set.StringSet
./executor/aggfuncs/func_avg.go:193:    valSet set.StringSet
./executor/aggfuncs/func_sum.go:43:     valSet set.StringSet
./executor/inspection_result.go:57:             set       set.StringSet
./expression/builtin_other.go:243:      hashSet set.StringSet
./expression/builtin_other.go:385:      hashSet set.StringSet
```

There are lots of places using `StringSet`, would they all have the same potential issue?

Reply:
> how about modifying the `stringmap`?

I'd rather not. The users of `stringmap` should have control of how they handle the memory.

Reply:
> There are lots of places using `StringSet`, would they all have the same potential issue?

This PR/issue aims to resolve the problem in package `executor/aggfuncs`, I'll take another look and create issues if there'are further problems and reply here.

Reply:
Others have no problem with my glance. And some comments are added.